File: src/DetectCapital.java
public class DetectCapital {
    public boolean detectCapitalUse(String word) {
        return isAllCaps(word) || isAllLowerCase(word, 0) || firstCharIsCapital(word);
    }

    private boolean isAllCaps(String string) {
        for (int index = 0 ; index < string.length() ; index++) {
            if (Character.isLowerCase(string.charAt(index))) {
                return false;
            }
        }
        return true;
    }

    private boolean isAllLowerCase(String string, int start) {
        for (int index = start ; index < string.length() ; index++) {
            if (Character.isUpperCase(string.charAt(index))) {
                return false;
            }
        }
        return true;
    }

    private boolean firstCharIsCapital(String string) {
        return Character.isUpperCase(string.charAt(0)) && isAllLowerCase(string, 1);
    }
}

File: python/detect_capital.py
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        return word.isupper() or word.islower() or word[0].isupper() and self.isLower(word, 1)

    def isLower(self, word: str, start: int) -> bool:
        for index in range(start, len(word)):
            if word[index].isupper():
                return False
        return True
